mlog/log.go

```diff
@@ -142,3 +142,12 @@ func (l *Logger) Error(message string, fields ...Field) {
 func (l *Logger) Critical(message string, fields ...Field) {
 	l.zap.Error(message, fields...)
 }
+
+// DoWithLogLevel allow for execution of code with the log level temporarily set to the parameter value,
+// after which the log level returns to its original value.
+func (l *Logger) DoWithLogLevel(level string, f func()) {
```

Comment:
Should we add a warning on this method not to use this outside of tests? Presumably it's not safe to rely on this in the presence of goroutines.